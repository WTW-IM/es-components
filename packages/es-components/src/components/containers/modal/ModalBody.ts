import styled from 'styled-components';

// Note: ModalBody relies on a parent (Modal) with ThemeProvider wrapping it
const ModalBody = styled.div`
  color: ${props => props.theme.colors.gray9};
  padding: 15px;

  .small & {
    @media (min-width: ${props => props.theme.screenSize.phone}) {
      margin-top: unset;
    }
  }

  .medium & {
    @media (min-width: ${props => props.theme.screenSize.tablet}) {
      margin-top: unset;
    }
  }

  .large & {
    @media (min-width: ${props => props.theme.screenSize.desktop}) {
      margin-top: unset;
    }
  }
`;

export default ModalBody;
