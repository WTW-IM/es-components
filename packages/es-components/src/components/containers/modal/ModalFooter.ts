import styled from 'styled-components';

// Note: ModalFooter relies on a parent (Modal) with ThemeProvider wrapping it
const ModalFooter = styled.div`
  align-items: baseline;
  display: flex;
  padding: 15px 15px 55px;
  text-align: right;

  button {
    width: auto;
  }

  @media (min-width: ${props => props.theme.screenSize.tablet}) {
    padding: 15px 15px 19px;
  }
`;

export default ModalFooter;
