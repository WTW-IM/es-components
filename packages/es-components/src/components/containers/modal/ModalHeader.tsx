import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import OrigDismissButton from '../../controls/DismissButton';
import { ModalContext } from './ModalContext';
import Heading from '../heading/Heading';
import { HeadingLevel } from 'es-components-shared-types';

const DismissButton = OrigDismissButton as React.ForwardRefExoticComponent<
  JSX.IntrinsicElements['button']
>;

// Note: ModalHeader relies on a parent (Modal) with ThemeProvider wrapping it
const Header = styled.div`
  align-items: baseline;
  background-color: ${props => props.theme.colors.white};
  border-bottom: 2px solid ${props => props.theme.brandColors.primary3};
  box-sizing: border-box;
  color: ${props => props.theme.colors.gray9};
  display: flex;
  font-size: 26px;
  justify-content: space-between;
  padding: 15px;

  .small & {
    @media (min-width: ${props => props.theme.screenSize.phone}) {
      position: static;
      width: unset;
    }
  }

  .medium & {
    @media (min-width: ${props => props.theme.screenSize.tablet}) {
      position: static;
      width: unset;
    }
  }

  .large & {
    @media (min-width: ${props => props.theme.screenSize.desktop}) {
      position: static;
      width: unset;
    }
  }
`;

const Title = styled(Heading)`
  margin: 0;
`;

const DismissModal = styled(DismissButton)`
  align-self: center;
  color: ${props => props.theme.colors.black};
  font-size: 18px;
`;

interface ModalHeaderProps {
  hideCloseButton: boolean;
  children: React.ReactNode;
  level: HeadingLevel;
}

function ModalHeader(props: ModalHeaderProps) {
  const { hideCloseButton, children, level, ...otherProps } = props;
  const { onHide, ariaId } = useContext(ModalContext);

  return (
    <Header {...otherProps}>
      <Title id={ariaId} level={level}>
        {children}
      </Title>

      {!hideCloseButton && <DismissModal onClick={onHide} />}
    </Header>
  );
}

ModalHeader.propTypes = {
  /** Specify whether the modal header should contain a close button */
  hideCloseButton: PropTypes.bool,
  children: PropTypes.any,
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  ref: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any })
  ])
};

ModalHeader.defaultProps = {
  hideCloseButton: false,
  children: undefined,
  level: 4,
  ref: null
};

export default ModalHeader;
