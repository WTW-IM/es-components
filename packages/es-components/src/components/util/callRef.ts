import React from 'react';

const callRef = <T, R extends React.ForwardedRef<T>>(ref: R, value: T) =>
  typeof ref === 'function' ? ref(value) : ref && (ref.current = value);

export default callRef;
