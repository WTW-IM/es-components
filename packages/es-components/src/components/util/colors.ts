import tinycolor, { ColorInput } from 'tinycolor2';

export function darken(color: ColorInput, percent: number) {
  return tinycolor(color).darken(percent).toHexString();
}

export function lighten(color: ColorInput, percent: number) {
  return tinycolor(color).lighten(percent).toHexString();
}

export function getTextColor(
  bgColor: ColorInput,
  darkestColor: ColorInput = 'black',
  lightestColor: ColorInput = 'white'
) {
  return tinycolor.readability(darkestColor, bgColor) > 10
    ? darkestColor
    : lightestColor;
}
