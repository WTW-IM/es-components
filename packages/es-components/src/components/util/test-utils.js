// eslint-env jest
import React from 'react';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render } from '@testing-library/react';

export function ThemeComponent(props) {
  return <ThemeProvider theme={viaTheme} {...props} />;
}

export function renderWithTheme(component) {
  return render(<ThemeComponent>{component}</ThemeComponent>);
}
