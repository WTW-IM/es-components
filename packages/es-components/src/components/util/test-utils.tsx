import React, { Profiler } from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render, RenderResult } from '@testing-library/react';

export function ThemeComponent(
  props: Omit<React.ComponentPropsWithoutRef<typeof ThemeProvider>, 'theme'>
) {
  return <ThemeProvider {...props} theme={viaTheme} />;
}

ThemeComponent.propTypes = {
  children: PropTypes.node
};

const profilerRender = (id: string, phase: string, actualDuration: number) => {
  if (actualDuration < 100) return;
  console.warn('Render took too long!', { id, phase, actualDuration });
};

export function renderWithTheme(component: React.ReactElement): RenderResult {
  const { ...viewUtils } = render(<ThemeComponent>{component}</ThemeComponent>);
  return {
    ...viewUtils,
    rerender: (rerenderComponent: React.ReactNode) =>
      viewUtils.rerender(
        <Profiler id="ThemeRender" onRender={profilerRender}>
          <ThemeComponent>{rerenderComponent}</ThemeComponent>
        </Profiler>
      )
  };
}

export const setClientWidth = (width: number) => {
  Object.defineProperty(document.documentElement, 'clientWidth', {
    writable: true,
    configurable: true,
    value: width
  });
};

export const setClientHeight = (height: number) => {
  Object.defineProperty(document.documentElement, 'clientHeight', {
    writable: true,
    configurable: true,
    value: height
  });
};
