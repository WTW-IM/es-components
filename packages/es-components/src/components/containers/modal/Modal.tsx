import React, { useCallback, useState, useEffect, useRef } from 'react';
import * as CSS from 'csstype';
import PropTypes from 'prop-types';
import {
  createGlobalStyle,
  css,
  DefaultTheme,
  ThemeProvider,
  ThemeProviderComponent
} from 'styled-components';
import { noop, debounce } from 'lodash';
import ReactModal from 'react-modal';
import tinycolor from 'tinycolor2';

import useUniqueId from '../../util/useUniqueId';
import callRef from '../../util/callRef';
import { useDisableBodyScroll } from '../../util/useDisableBodyScroll';
import { useRootNodeLocator, RootNode } from '../../util/useRootNode';
import { ModalContext } from './ModalContext';
import Header from './ModalHeader';
import Body from './ModalBody';
import Footer from './ModalFooter';
import useTopZIndex from '../../../hooks/useTopZIndex';

export type ModalTheme = DefaultTheme & {
  modalHeight: number;
  windowHeight: number;
  portalClassName: string;
};

type HTMLRef = React.ForwardedRef<HTMLElement> | undefined;
type BackdropValue = boolean | 'static';

const modalSize = {
  small: '300px',
  medium: '600px',
  large: '900px'
} as const;

export type ModalSize = keyof typeof modalSize;

export type ModalProps = Override<
  Omit<ReactModal.Props, 'isOpen'>,
  {
    show?: boolean;
    size?: ModalSize;
    parentSelector?: () => RootNode;
    className?: string;
    animation?: boolean;
    backdrop?: BackdropValue;
    children?: React.ReactNode;
    escapeExits?: boolean;
    onEnter?: () => void;
    onExit?: () => void;
    onHide?: () => void;
    overlayRef?: HTMLRef;
    contentRef?: HTMLRef;
  }
>;

const animationTimeMs = 300 as number;

const getModalMargin = (windowHeight: number, modalHeight: number) => {
  const fullMargin = windowHeight - modalHeight;
  const thirdTopMargin = fullMargin / 3;
  return Math.max(thirdTopMargin, 50);
};

const getAlphaColor = (color: CSS.Property.Color, alpha: number) => {
  const colorObj = tinycolor(color);
  colorObj.setAlpha(alpha);
  return colorObj.toRgbString();
};

const ModalStyles = createGlobalStyle<{
  showBackdrop: boolean;
  showAnimation?: boolean;
  theme: ModalTheme;
  topIndex: number;
}>`
${({
  theme: {
    portalClassName,
    colors,
    font,
    screenSize,
    windowHeight,
    modalHeight
  },
  showBackdrop,
  topIndex,
  showAnimation
}) => css`
  .${portalClassName} {
    .background-overlay {
      bottom: 0;
      background-color: ${getAlphaColor(colors.gray9, 0)};

      left: 0;
      max-height: 100%;
      overflow-y: auto;
      position: fixed;
      right: 0;
      top: 0;
      transition: background-color ${animationTimeMs / 2}ms linear;
      z-index: ${topIndex};
      -webkit-overflow-scrolling: touch;

      &.ReactModal__Overlay--after-open {
        background-color: ${() => {
          if (!showBackdrop) return '';

          return getAlphaColor(colors.gray9, 0.5);
        }};

        &.ReactModal__Overlay--before-close {
          background-color: ${getAlphaColor(colors.gray9, 0)};
        }
      }
    }

    .modal-content {
      background-clip: padding-box;
      background-color: ${colors.white};
      border-radius: 3px;
      box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
      font-family: 'Source Sans Pro', 'Segoe UI', Segoe, Calibri, Tahoma,
        sans-serif;
      font-size: ${font.baseFontSize};
      margin: 0.5rem;
      outline: 0;
      opacity: 0;
      position: absolute;
      left: 0;
      transform: translateY(0);
      transition: transform ${animationTimeMs}ms ease-out
          ${animationTimeMs / 4}ms,
        opacity ${animationTimeMs}ms ease-out ${animationTimeMs / 4}ms;
      ${showAnimation
        ? `
            transform: translateY(-50px);
          `
        : ''};

      &.ReactModal__Content--after-open {
        opacity: 1;
        ${showAnimation
          ? `
              transform: translateY(0);
            `
          : ''};

        &.ReactModal__Content--before-close {
          opacity: 0;
          ${showAnimation
            ? `
                transform: translateY(-50px);
              `
            : ''};
        }
      }

      &.small {
        @media (min-width: ${screenSize.phone}) {
          position: relative;
          margin: ${getModalMargin(windowHeight, modalHeight)}px auto;
          width: ${modalSize.small};
        }
      }

      &.medium {
        @media (min-width: ${screenSize.tablet}) {
          position: relative;
          margin: ${getModalMargin(windowHeight, modalHeight)}px auto;
          width: ${modalSize.medium};
        }
      }

      &.large {
        @media (min-width: ${screenSize.desktop}) {
          position: relative;
          margin: ${getModalMargin(windowHeight, modalHeight)}px auto;
          width: ${modalSize.large};
        }
      }
    }
  }
`}`;

const ModalThemeProvider = ThemeProvider as unknown as ThemeProviderComponent<
  ModalTheme,
  DefaultTheme
>;

function Modal({
  animation = true,
  backdrop = true,
  children,
  escapeExits,
  onEnter,
  onExit,
  onHide,
  show,
  size,
  parentSelector,
  className,
  overlayRef,
  contentRef,
  ...other
}: ModalProps) {
  const ariaId = useUniqueId(other.id as string);
  const portalClassName = `ReactModalPortal-${useUniqueId()}`;
  const [rootNode, RootNodeLocator] = useRootNodeLocator(document.body);
  const [modalHeight, setModalHeight] = useState(300);
  const [shouldShow, setShouldShow] = useState(show);
  const [windowHeight, setWindowHeight] = useState(window.innerHeight);
  const modalRef = useRef<HTMLElement | null>(null);
  const innerContentRef = useRef<HTMLElement | null>(null);
  const getTopIndex = useTopZIndex();
  const setContentRef = useCallback(
    (contentElement: HTMLElement) => {
      callRef(contentRef, contentElement);
      innerContentRef.current = contentElement;
    },
    [contentRef]
  );

  const modalHeightRef = useCallback(
    (modalElement: HTMLElement) => {
      callRef(overlayRef, modalElement);
      let newHeight = 0;
      if (modalElement) {
        modalRef.current = modalElement;
        newHeight = modalElement.offsetHeight;
      }
      if (innerContentRef.current) {
        const contRefHeight = innerContentRef.current.offsetHeight;
        if (contRefHeight < newHeight) newHeight = contRefHeight;
      }
      setModalHeight(newHeight);
    },
    [overlayRef]
  );

  const rootNodeSelector = useCallback(() => rootNode, [rootNode]);
  const modalParentSelector = parentSelector || rootNodeSelector;

  useEffect(() => {
    if (!animation || show) {
      setShouldShow(show);
      return noop;
    }

    let mounted = true;
    setTimeout(() => {
      if (!mounted) return;

      setShouldShow(false);
    }, animationTimeMs);

    return () => {
      mounted = false;
    };
  }, [show, animation]);

  useEffect(() => {
    const setHeight = debounce(() => setWindowHeight(window.innerHeight), 100);
    window.addEventListener('resize', setHeight);
    return () => window.removeEventListener('resize', setHeight);
  }, []);

  useEffect(() => {
    if (modalRef.current && modalRef.current.scroll) {
      modalRef.current.scroll(0, 0);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [modalRef.current]);

  useDisableBodyScroll(show);

  const shouldCloseOnOverlayClick = backdrop === 'static' ? false : backdrop;

  return (
    <ModalThemeProvider
      theme={theme => ({
        ...theme,
        modalHeight,
        windowHeight,
        portalClassName
      })}
    >
      <RootNodeLocator />
      {shouldShow ? (
        <ModalStyles
          showAnimation={animation}
          showBackdrop={Boolean(backdrop)}
          topIndex={getTopIndex()}
        />
      ) : null}
      <ModalContext.Provider value={{ onHide: onHide || noop, ariaId }}>
        <ReactModal
          portalClassName={portalClassName}
          className={`${className || ''} ${size || 'medium'} modal-content`}
          overlayClassName="background-overlay"
          closeTimeoutMS={animation ? animationTimeMs : undefined}
          aria={{
            labelledby: ariaId
          }}
          shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
          onRequestClose={onHide}
          onAfterOpen={onEnter}
          onAfterClose={onExit}
          shouldCloseOnEsc={escapeExits}
          parentSelector={modalParentSelector as () => HTMLElement}
          contentRef={setContentRef}
          overlayRef={modalHeightRef}
          {...{ ...(other || {}), isOpen: Boolean(show) }}
        >
          {children}
        </ReactModal>
      </ModalContext.Provider>
    </ModalThemeProvider>
  );
}

const badProp = (
  props: Record<string, unknown>,
  propName: string,
  componentName: string
) => {
  if (props[propName]) {
    return new Error(
      `Invalid ReactModal prop '${propName}' supplied to '${componentName}'.`
    );
  }
};

export const propTypes = {
  ...ReactModal.propTypes,
  isOpen: badProp,
  overlayClassName: badProp,
  closeTimeoutMS: badProp,
  aria: badProp,
  onRequestClose: badProp,
  onAfterOpen: badProp,
  onAfterClose: badProp,
  shouldCloseOnEsc: badProp,
  /** Open and close the Modal with transitions. */
  animation: PropTypes.bool,
  /**
   * Include a backdrop component. Specify 'static' for a backdrop that doesn't
   * trigger an "onHide" when clicked.
   */
  backdrop: PropTypes.oneOf(['static', true, false]),
  /**
   * Usually contains a Modal.Header, Modal.Body, and Modal.Footer
   * but can contain any content
   */
  children: PropTypes.any,
  /** Set whether the Escape key exits the modal */
  escapeExits: PropTypes.bool,
  /** Callback fired when the Modal transitions in */
  onEnter: PropTypes.func,
  /** Callback fired when the modal transitions out */
  onExit: PropTypes.func,
  /**
   * A callback fired when the header closeButton or non-static backdrop is
   * clicked. Required if either are specified.
   */
  onHide: PropTypes.func,
  /** When `true` The modal will show itself. */
  show: PropTypes.bool,
  /** Sets the size of the modal */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /** Selects the parent */
  parentSelector: PropTypes.func,
  className: PropTypes.string,
  overlayRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any })
  ]),
  contentRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any })
  ])
};

export const defaultProps = {
  animation: true,
  backdrop: true,
  escapeExits: true,
  onEnter: noop,
  onExit: noop,
  onHide: noop,
  show: false,
  size: 'medium',
  children: undefined,
  parentSelector: null,
  className: null,
  overlayRef: null,
  contentRef: null
};

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

Modal.Header = Header;
Modal.Body = Body;
Modal.Footer = Footer;

export default Modal;
