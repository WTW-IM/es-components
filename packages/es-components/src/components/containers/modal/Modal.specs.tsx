/* eslint-env jest */

import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react';

import Modal from './Modal';
import ModalHeader from './ModalHeader';
import ModalBody from './ModalBody';
import ModalFooter from './ModalFooter';
import { renderWithTheme } from '../../util/test-utils';

jest.mock('../../util/generateAlphaName', () => () => 'abcdef');

type ModalType = typeof Modal;

function renderModal(
  modalProps: Parameters<ModalType>[0],
  children: React.ReactNode
) {
  return renderWithTheme(
    <Modal ariaHideApp={false} show {...{ ...modalProps, children }} />
  );
}

function renderBasicModal(modalProps: Parameters<ModalType>[0]) {
  return renderModal(
    modalProps,
    <>
      <ModalHeader>Header</ModalHeader>
      <ModalBody>Body</ModalBody>
      <ModalFooter>Footer</ModalFooter>
    </>
  );
}

it('invokes functions when triggered', async () => {
  const onEnter = jest.fn();
  const onHide = jest.fn();

  const utils = renderBasicModal({
    onEnter,
    onHide
  });

  if (utils === null) {
    return;
  }
  const { findByRole } = utils;

  await waitFor(() => {
    expect(onEnter).toHaveBeenCalled();
  });

  fireEvent.click(await findByRole('button', { name: 'Close', exact: false }));

  expect(onHide).toHaveBeenCalled();
});

it('renders different modal sections', async () => {
  const onEnter = jest.fn();
  const onHide = jest.fn();

  const { getByText } = renderBasicModal({ onEnter, onHide });

  await waitFor(() => {
    expect(onEnter).toHaveBeenCalled();
  });

  expect(getByText('Header')).toMatchSnapshot();
  expect(getByText('Body')).toMatchSnapshot();
  expect(getByText('Footer')).toMatchSnapshot();
});

it('hides dismiss button when hideCloseButton is true', () => {
  const { getByText } = renderModal(
    {},
    <>
      <ModalHeader hideCloseButton>Header</ModalHeader>
      <ModalBody>Body</ModalBody>
      <ModalFooter>Footer</ModalFooter>
    </>
  );
  expect(getByText('Header').parentElement?.querySelector('button')).toBeNull();
});

it('passes through class names', () => {
  const { container } = renderBasicModal({ className: 'myclass' });
  const element = container.parentElement?.querySelector('.myclass');
  expect(element).not.toBeNull();
});

describe('when ESC is pressed', () => {
  it('invokes onHide by default', async () => {
    const onHide = jest.fn();

    const { queryByText } = renderBasicModal({ onEnter: jest.fn(), onHide });

    const view = queryByText('Header');
    if (view === null) {
      return;
    }
    fireEvent.keyDown(view, { keyCode: 27 });
    await waitFor(() => {
      expect(onHide).toHaveBeenCalled();
    });
  });

  it('does not invoke onHide when escapeExits is false', async () => {
    const onHide = jest.fn();
    const { container } = renderBasicModal({ onHide, escapeExits: false });

    fireEvent.keyDown(container, { keyCode: 27 });
    await waitFor(() => {
      expect(onHide).not.toHaveBeenCalled();
    });
  });
});
