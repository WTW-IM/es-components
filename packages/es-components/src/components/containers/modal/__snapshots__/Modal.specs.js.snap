// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders different modal sections 1`] = `
.c0 {
  color: inherit;
  font-family: 'Source Sans Pro','Segoe UI',Segoe,Calibri,Tahoma,sans-serif;
  font-size: 25.92px;
  font-weight: 300;
  line-height: 1.1;
  margin-bottom: 0.45em;
  margin-top: 0;
}

.c0 small {
  font-size: 75%;
  line-height: 1;
}

.c1 {
  margin: 0;
}

@media (min-width:768px) {
  .c0 {
    font-size: 25.92px;
  }
}

@media (min-width:480px) {

}

@media (min-width:768px) {

}

@media (min-width:900px) {

}

@media (min-width:480px) {

}

@media (min-width:768px) {

}

@media (min-width:900px) {

}

@media (min-width:768px) {

}

<h4
  class="c0 c1"
  id="abcdef"
>
  Header
</h4>
`;

exports[`renders different modal sections 2`] = `
.c1 {
  color: #444;
  padding: 15px;
}

@media (min-width:768px) {

}

@media (min-width:480px) {

}

@media (min-width:768px) {

}

@media (min-width:900px) {

}

@media (min-width:480px) {
  .small .c0 {
    margin-top: unset;
  }
}

@media (min-width:768px) {
  .medium .c0 {
    margin-top: unset;
  }
}

@media (min-width:900px) {
  .large .c0 {
    margin-top: unset;
  }
}

@media (min-width:768px) {

}

<div
  class="c0 c1"
>
  Body
</div>
`;

exports[`renders different modal sections 3`] = `
.c0 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px 15px 55px;
  text-align: right;
}

.c0 button {
  width: auto;
}

@media (min-width:768px) {

}

@media (min-width:480px) {

}

@media (min-width:768px) {

}

@media (min-width:900px) {

}

@media (min-width:480px) {

}

@media (min-width:768px) {

}

@media (min-width:900px) {

}

@media (min-width:768px) {
  .c0 {
    padding: 15px 15px 19px;
  }
}

<div
  class="c0"
>
  Footer
</div>
`;
